libs/crypto/tests/unit/mcl_dkg_tests.cpp

```diff
@@ -0,0 +1,271 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "crypto/mcl_dkg.hpp"
+#define MCLBN_FP_UNIT_SIZE 4
+
+#include "gtest/gtest.h"
+
+#include <cstdint>
+#include <iostream>
+#include <ostream>
+
+using namespace fetch::crypto::mcl;
+using namespace fetch::byte_array;
+
+TEST(MclTests, BaseMcl)
+{
+  const char *aa = "12723517038133731887338407189719511622662176727675373276651903807414909099441";
+  const char *ab = "4168783608814932154536427934509895782246573715297911553964171371032945126671";
+  const char *ba = "13891744915211034074451795021214165905772212241412891944830863846330766296736";
+  const char *bb = "7937318970632701341203597196594272556916396164729705624521405069090520231616";
+
+  mcl::bn::initPairing();
+  mcl::bn::G2 Q(mcl::bn::Fp2(aa, ab), mcl::bn::Fp2(ba, bb));
+  mcl::bn::G1 P(-1, 1);
+
+  // Comparison
+  {
+    mcl::bn::G2 Q0;
```

Comment:
Perhaps I should get Troels to scan this as I am not familiar with these operations

Reply:
They are addition and multiplication operations on the curve. This test is just a basic test to make sure mcl is working properly. We can remove it if we want.